### PR TITLE
fix: show anonymous user for empty nicknames

### PR DESCRIPTION
**Pull request by sunner**
## Summary

- change the shared empty-nickname label to `Anonymous User` and align the Chinese and French translations
- make the credit-notification managed list reuse the shared user label instead of its local `No nickname` copy
- trim managed-list nicknames so whitespace-only values also use the shared fallback
- show the same shared fallback in admin and teacher follow-up lists, follow-up detail drawers, and the admin rating list
- keep guest rating rows on their dedicated `Guest` label instead of adding a second anonymous-user label
- preserve real nicknames on rating rows that have no phone or email
- preserve stored nicknames that happen to match a localized fallback label instead of treating display text as a data sentinel
- remove the obsolete locale key, regenerate i18n types, and assert the rendered `Anonymous User` label in the regression test

## Why

Empty nicknames used inconsistent placeholder copy across the product. The shared label also read like a state description instead of a display name, and whitespace-only values could still render as blank.

This change gives the relevant admin nickname fields a consistent, name-like fallback while removing the remaining local exceptions. Missing names are determined from the source nickname, so a real nickname such as `Anonymous User` remains visible in every locale.

## Validation

- `npm run test -- --runInBand src/app/admin/operations/credit-notifications/page.test.tsx` (25 tests passed)
- focused course detail, follow-up, and rating suites (52 tests passed)
- exact-path follow-up and rating regression suites (30 tests passed)
- exact-path rating regression suite after the guest edge-case fixes (11 tests passed)
- `npm run type-check`
- `npm run lint`
- `python3 scripts/check_translations.py`
- `python3 scripts/check_translation_usage.py --fail-on-unused`
- repository pre-commit and commit-message hooks

### CODE REVIEW

File: src/cook-web/src/app/admin/operations/credit-notifications/page.test.tsx [hunk 1]
```diff
@@ -1081,7 +1081,7 @@ describe('AdminOperationCreditNotificationsPage', () => {
         'module.operationsCreditNotifications.config.fields.blockedCreatorList',
       ).length,
     ).toBeGreaterThan(1);
-    expect(screen.getByText('Creator One')).toBeInTheDocument();
+    expect(screen.getByText('module.user.defaultUserName')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/cook-web/src/app/admin/operations/credit-notifications/page.test.tsx]: Fixed in f9eb3cfb3. The test mock now translates the shared key to Anonymous User, and the assertion verifies the rendered user-visible label. The focused page suite passes all 25 tests.

File: src/cook-web/src/app/admin/operations/credit-notifications/CreditNotificationManagedListDialog.tsx [hunk 2]
```diff
@@ -92,10 +92,7 @@ export function CreditNotificationManagedListDialog({
                         {contactValue}
                       </p>
                       <p className='mt-0.5 truncate text-xs text-muted-foreground'>
-                        {item.nickname ||
-                          t(
-                            'module.operationsCreditNotifications.config.listDialog.emptyNickname',
-                          )}
+                        {item.nickname || t('module.user.defaultUserName')}
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/cook-web/src/app/admin/operations/credit-notifications/CreditNotificationManagedListDialog.tsx]: Fixed in 4902f3959. Managed-list nicknames are now trimmed before fallback selection, and the regression fixture uses a whitespace-only nickname. The focused page suite passes all 25 tests.

File: src/i18n/en-US/modules/user.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "__namespace__": "module.user",
   "confirmLogoutContent": "Are you sure you want to logout?",
   "confirmLogoutTitle": "Confirm Logout",
-  "defaultUserName": "Default Name",
+  "defaultUserName": "Anonymous User",
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/i18n/en-US/modules/user.json:5]: Fixed in a118051ab. Course detail, follow-up list/detail, and rating views now derive missing-name state only from the source nickname and no longer compare stored nicknames with localized fallback text. Added regression coverage showing a real nickname of Anonymous User remains visible; 52 focused tests and type-check passed.

File: src/cook-web/src/app/admin/operations/[shifu_bid]/follow-ups/FollowUpDetailSheet.tsx [hunk 4]
```diff
@@ -231,11 +229,8 @@ export default function FollowUpDetailSheet({
   );
   const nickname = useMemo(() => {
     const normalizedNickname = basicInfo?.nickname?.trim() || '';
-    if (!normalizedNickname || normalizedNickname === defaultUserName) {
-      return emptyValue;
-    }
-    return normalizedNickname;
-  }, [basicInfo?.nickname, defaultUserName, emptyValue]);
+    return normalizedNickname || emptyValue;
+  }, [basicInfo?.nickname, emptyValue]);
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/cook-web/src/app/admin/operations/[shifu_bid]/follow-ups/FollowUpDetailSheet.tsx]: Fixed in a458a6c19. Empty and whitespace-only nicknames now use the shared `module.user.defaultUserName` label in admin and teacher follow-up lists, both follow-up detail drawers, and the admin rating list. The fallback is selected only when the source nickname is empty, so real nicknames remain unchanged. All 30 focused follow-up/rating tests, type-check, lint, and commit hooks passed.

File: src/cook-web/src/app/admin/operations/[shifu_bid]/ratings/page.tsx [hunk 5]
```diff
@@ -450,13 +450,8 @@ export default function AdminOperationCourseRatingsPage() {
...
   const resolveUserSecondary = useCallback(
-    (item: AdminOperationCourseRatingItem) => {
-      const nickname = item.nickname?.trim() || '';
-      if (!nickname || nickname === defaultUserName) {
-        return '';
-      }
-      return nickname;
-    },
+    (item: AdminOperationCourseRatingItem) =>
+      item.nickname?.trim() || defaultUserName,
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/cook-web/src/app/admin/operations/[shifu_bid]/ratings/page.tsx:454]: Fixed in ac00e083f. Guest rating rows now suppress the secondary anonymous-user fallback when the dedicated Guest label is already the primary identity. The existing guest regression now asserts that `Anonymous User` is absent; all 10 rating tests, type-check, lint, and commit hooks passed.

File: src/cook-web/src/app/admin/operations/[shifu_bid]/ratings/page.tsx [hunk 6]
```diff
@@ -969,10 +964,11 @@ export default function AdminOperationCourseRatingsPage() {
                                   contactMode,
                                   emptyValue,
                                 });
-                                const secondaryAccount =
-                                  resolveUserSecondary(item);
                                 const isGuestAccount =
                                   !item.mobile?.trim() && !item.email?.trim();
+                                const secondaryAccount = isGuestAccount
+                                  ? ''
+                                  : resolveUserSecondary(item);
```
[the remark this answers is not in the document]
Comment (sunner) [reply src/cook-web/src/app/admin/operations/[shifu_bid]/ratings/page.tsx]: Fixed in 8f884d424. The guest-row gate now suppresses the secondary label only when both contact information and the source nickname are empty. Contactless users with a real nickname keep that nickname visible. Added a dedicated regression test; all 11 rating tests, type-check, lint, and commit hooks passed.
